lib/playground.dart

```diff
@@ -606,35 +609,69 @@ class Playground implements GistContainer, GistController {
     }
   }
 
-  void _handleRun() {
+  void _handleRun() async {
     ga.sendEvent('main', 'run');
     runButton.disabled = true;
     overlay.visible = true;
 
     Stopwatch compilationTimer = Stopwatch()..start();
 
-    var input = CompileRequest()..source = context.dartSource;
-    dartServices
-        .compile(input)
-        .timeout(longServiceCallTimeout)
-        .then((CompileResponse response) {
-      ga.sendTiming('action-perf', 'compilation-e2e',
-          compilationTimer.elapsedMilliseconds);
-
-      _autoSwitchOutputTab();
-      _clearOutput();
-
-      return executionService.execute(
-          _context.htmlSource, _context.cssSource, response.result);
-    }).catchError((e) {
+    final CompileRequest compileRequest = CompileRequest()
+      ..source = context.dartSource;
+
+    try {
+      if (_useDDC) {
+        final CompileDDCResponse response = await dartServices
+            .compileDDC(compileRequest)
+            .timeout(longServiceCallTimeout);
+
+        ga.sendTiming(
+          'action-perf',
+          'compilation-e2e',
+          compilationTimer.elapsedMilliseconds,
+        );
+
+        _autoSwitchOutputTab();
+        _clearOutput();
+
+        return executionService.execute(
+          _context.htmlSource,
+          _context.cssSource,
+          response.result,
+          modulesBaseUrl: response.modulesBaseUrl,
+        );
+      } else {
+        final CompileResponse response = await dartServices
+            .compile(compileRequest)
+            .timeout(longServiceCallTimeout);
+
+        ga.sendTiming(
+          'action-perf',
+          'compilation-e2e',
+          compilationTimer.elapsedMilliseconds,
+        );
+
+        _autoSwitchOutputTab();
+        _clearOutput();
+
+        return executionService.execute(
+          _context.htmlSource,
+          _context.cssSource,
+          response.result,
+        );
+      }
+    } catch (e) {
       ga.sendException('${e.runtimeType}');
-      if (e is DetailedApiRequestError) e = e.message;
+      dynamic message = e;
```

Comment:
This is a nit (and I know you're just formatting these lines rather than writing them), but it took me a second to understand what's going on here.

Could this alternate version of 665-668 work and be more readable?

```
final message = (e is DetailedApiRequestError) ? e.message : '$e';
```

Reply:
👍 sounds good; will update